audio-worklet/bitcrusher-test.html

```diff
@@ -16,171 +16,184 @@
-    	          
-    	          if (i < bufferLength) {
-    	            console.assert(
-    	                bitcrusherOutput[i] == 0,
-    	                'Bitcrusher sample at ' + i + 'expected to be 0 but it'
-    	                + ' was ' + bitcrusherOutput[i]);
-    	          }
-    	          console.assert(
-    	              diff < permittedSampleError,
-    	              'Bitcrushed sample at ' + i + ' is ' + crushedSample + 
-    	              ' but ' + originalSample + ' for oscillator');
-    	        }
-    	      });
-    	}
-
-    	/**
-    	 * Verify that bitcrusher handles illegal input correctly.
-    	 */
-    	function testIllegalInput() {
-    	  try {
-    	    let bitcrusher = new Bitcrusher(0);
-    	    console.error('Bitcrusher did not throw illegal context error.');
-    	  }
-    	  catch (error) {}
-    	}
-
-    	/**
-    	 * Verify that bitcrusher applies sample reduction on multiple channels of 
-    	 * input, and that sample reduction works across multiple buffers (which 
-    	 * requires bitcrusher memory.)
-    	 */
-    	function testMultipleChannels() {
-    	  let channels = 2;
-    	  let context = new OfflineAudioContext(channels, 44100, 44100);
-    	  // Reduction should not be a factor of buffersize for full coverage.
-    	  const reduction = 7; 
-    	  const bitDepth = 24;
-    	  const bufferSize = 512;
-    	  let oscillator = new OscillatorNode(context);
-    	  let splitter = new ChannelSplitterNode(context);
-
-    	  let bitcrusher = new Bitcrusher(context, {
-    	    bufferSize: bufferSize,
-    	    channels: channels,
-    	    bitDepth: bitDepth,
-    	    reduction: reduction
-    	  })
-
-    	  oscillator.connect(splitter).connect(bitcrusher.input);
-    	  bitcrusher.output.connect(context.destination); 
-    	  oscillator.start();
-    	  
-    	  // When audio buffer is ready, verify bitcrushed samples are unaltered.
-    	  context.startRendering().then((buffer) => {      
-    	      let last = buffer.getChannelData(0)[0];
-    	      for (let i = 1; i < buffer.getChannelData(0).length; i++) {
-    	        for (let c = 0; c < channels; c++) {
-    	          console.assert(
-    	              buffer.getChannelData(c)[i] === last, 
-    	              'At iteration ' + i + ', the value at channel ' + c + ', ' +
-    	                  buffer.getChannelData(c)[i] + ' did not match ' +
-    	                  ' expectation, ' + last);
-    	          if (i % reduction === 0) last = buffer.getChannelData(c)[i];
-    	        }
-    	     } 
-    	  });  
-    	}
+      testSampleReduction();
+      testBitcrusherInAudioContext();
+      testMultipleChannels();
+      testIllegalInput();
+
+      /**
+       * Populate array with 0:99 and verify the bitcrusher holds an array
+       * where the number of consecutive equal numbers is controlled by the 
+       * variable factor. This verfies the sample rate reduction component
```

Comment:
Typo: "verfies"

The first sentence should say what the test is, and the remaining sentence(s) can describe how you're testing this.